examples/bidder/http_bidder_test.cpp

```diff
@@ -95,49 +103,63 @@ int main(int argc, char *argv[]) {
+        reply << http::server::reply::flush("");
+    };
+    auto auction_async_f = [&bid_handler](http::server::reply &reply, BidRequest & bid_request) -> bool {
+        return bid_handler.handle_auction_async(reply, bid_request);
+    };
+    const decision_exchange_type::decision_tree_type decision_tree = {{
+        {static_cast<int> (decision_codes_type::USER_DATA),
+            {request_user_data_f, static_cast<int> (decision_codes_type::AUCTION_ASYNC), static_cast<int> (decision_codes_type::NO_BID),}},
+                
+        {static_cast<int> (decision_codes_type::NO_BID),
+            {no_bid_f, decision_exchange_type::decision_action::EXIT, decision_exchange_type::decision_action::EXIT}},
+                
+        {static_cast<int> (decision_codes_type::AUCTION_ASYNC),
+            {auction_async_f, decision_exchange_type::decision_action::EXIT, decision_exchange_type::decision_action::EXIT}}
+   }};
+   decision_exchange_type decision_exchange(decision_tree);
+    
     bid_handler    
         .logger([](const std::string &data) {
             //LOG(debug) << "bid request=" << data ;
         })
         .error_logger([](const std::string &data) {
             LOG(debug) << "bid request error " << data ;
         })
-
         .auction_async([&](const BidRequest &request) {
-
-            thread_local vanilla::BidderSelector<> selector(caches);
-            BidResponse response;
-            for(auto &imp : request.imp) {
-                if(auto ad = selector.select(request, imp)) {
-                   boost::uuids::uuid bidid = uuid_generator();
-                   response.bidid = boost::uuids::to_string(bidid);
-                   if (request.cur.size()) {
-                       response.cur = request.cur[0];
-                   } else if (imp.bidfloorcur.length()) {
-                       response.cur = imp.bidfloorcur; // Just return back
-                   }
-                   Bid bid;
-                   bid.id = boost::uuids::to_string(bidid); // TODO check documentation 
-                   // Is it the same as response.bidid?
-                   // Wrong filling type
-                   bid.impid = imp.id;
-                   bid.price = ad->max_bid_micros / 1000000.0; // Not micros?
-                   bid.w = ad->width;
-                   bid.h = ad->height;
-                   bid.adm = ad->code;
-                   bid.adid = ad->ad_id;
-                   if (response.seatbid.size() == 0) {
-                      SeatBid seatbid;
-                      seatbid.bid.push_back(bid);
-                      response.seatbid.push_back(seatbid);
-                   } else {
-                      response.seatbid.back().bid.push_back(bid);
-                   }
-                }
-            }
-            return response;
-        });
+            thread_local vanilla::ResponseBuilder<BidderConfig> response_builder(caches);
```

Comment:
Should we rename ResponseBuilder<T> to Bidder<T> ? Because it uses cache selector and filters and creates response. From reading this part of code it's hard to tell where the actual bidding occurs.
So this could  be potential our bidder class?

```
thread_local vanilla::Bidder<BidderConfig> bidder(caches);
return bidder.process(request);
```